website/src/content/docs/docs/howtos/Generate client libraries/10versioning.mdx

```diff
@@ -464,6 +464,177 @@ func (client *ServiceClient) StableFunctionality(ctx context.Context, stableMode
 
 </ClientTabs>
 
+### New model/operation added in new api version
+
+You can use `@added` decorator to indicate when a specific Api feature was first introduced.
+When applied to models, operations, or parameters, `@added` marks the Api Version at which these elements became available. This prevents confusing error messages for customers using older version clients.
+
+#### New operation added in new api version
+
+In this example, `createUserProfile` is marked with `@added(Versions.v2024_02_01)`, meaning this operation only becomes available starting from the `2024-02-01` version. Elements without the `@added` decorator, like `getUserInfo`, are available across all defined versions.
+
+<ClientTabs>
+
+<ClientTabItem lang="typespec">
+
+```typespec
+import "@typespec/versioning";
+import "@typespec/http";
+
+using Versioning;
+using Http;
+
+@versioned(Versions)
+@service
+namespace My.Service;
+
+enum Versions {
+  v2024_01_01: "2024-01-01",
+  v2024_02_01: "2024-02-01",
+}
+
+model UserInfo {
+  username: string;
+}
+
+model UserProfile {
+  profileName: string;
+}
+
+@route("/users/info")
+op getUserInfo(@body userInfo: UserInfo): void;
+
+@added(Versions.v2024_02_01)
+@route("/users/profile")
+op createUserProfile(@body userProfile: UserProfile): void;
+```
+
+</ClientTabItem>
+
+```python
+# change does not infect generated model
+class UserProfile(_Model):
+
+    profile_name: str = rest_field(name="profileName", visibility=["read", "create", "update", "delete", "query"])
+    profile_details: "_models.ProfileDetails" = rest_field(
+        name="profileDetails", visibility=["read", "create", "update", "delete", "query"]
+    )
+...
+
+# operation was first introduced from Api Version 2024-02-01, and is unreachable to client with Api Version 2024_01_01
+@distributed_trace
+@api_version_validation(
+    method_added_on="2024-02-01",
+    api_versions_list=["2024-02-01"],
+)
+def create_user_profile(
+    self, user_profile: Union[_models.UserProfile, JSON, IO[bytes]], **kwargs: Any
+) -> None:
+...
+```
+
+```csharp
+// TODO
+```
+
+```typescript
+// TODO
+```
+
+```java
+// TODO
+```
+
+```go
+// TODO
+```
+
+</ClientTabs>
+
+#### New model property added in new api version
+
+In this example, the `ProfileDetails` model is marked with `@added(Versions.v2024_02_01)`, meaning this model as well as the property `profileDetails` of `UserProfile` only becomes available starting from the `2024-02-01` version, while other parameters of `UserProfile`, are available across all defined versions.
+
+<ClientTabs>
+
+<ClientTabItem lang="typespec">
+
+```typespec
+import "@typespec/versioning";
+import "@typespec/http";
+
+using Versioning;
+using Http;
+
+@versioned(Versions)
+@service
+namespace My.Service;
+
+enum Versions {
+  v2024_01_01: "2024-01-01",
+  v2024_02_01: "2024-02-01",
+}
+
+model UserInfo {
+  username: string;
+}
+
+model UserProfile {
+  profileName: string;
+
+  @added(Versions.v2024_02_01)
+  profileDetails: ProfileDetails;
```

Comment:
Add a required property is breaking change.

Reply:
Yeah, but not unacceptable I understand.

Reply:
Yeah, not unacceptable.

Just for me, when good behavior and bad behavior have same cost, I'd choose good behavior...